R/pkg/R/sparkR.R

```diff
@@ -373,8 +373,13 @@ sparkR.session <- function(
     overrideEnvs(sparkConfigMap, paramMap)
   }
 
+  deployMode <- ""
+  if (exists("spark.submit.deployMode", envir = sparkConfigMap)) {
+    deployMode <- sparkConfigMap[["spark.submit.deployMode"]]
+  }
```

Comment:
you could instead just do `deployMode <- sparkConfigMap[["spark.submit.deployMode"]]`
it will be set to NULL if the key is not set.